in header is _PAM_LEGACY_NONCONST used and in source RT_OS_SOLARIS for that

--- VirtualBox-7.1.10/Config.kmk.orig
+++ VirtualBox-7.1.10/Config.kmk
@@ -2103,7 +2103,7 @@
 DEFS.openbsd     = RT_OS_OPENBSD
 DEFS.os2         = RT_OS_OS2
 DEFS.os-agnostic = RT_OS_AGNOSTIC
-DEFS.solaris     = RT_OS_SOLARIS _FILE_OFFSET_BITS=64
+DEFS.solaris     = RT_OS_SOLARIS _FILE_OFFSET_BITS=64 _PAM_LEGACY_NONCONST
 DEFS.win         = RT_OS_WINDOWS __WIN__
 DEFS.win.x86     = __WIN32__
 DEFS.win.amd64   = __WIN64__
